### ISSUE TITLE
TS 3.7: optional chaining output is introducing a local variable

### ISSUE THREAD

**Issue by bpasero** [microsoft/TypeScript#34594]
**TypeScript Version:**  3.7.x-dev.201xxxxx

**Search Terms:**
optional chaining

**Code**

```ts
function isTrue(something: { value: boolean} | undefined): boolean {
	return something?.value === true; 
}
```

**Expected behavior:**
Output:
```js
"use strict";
function isTrue(something) {
    return (something === null || something === void 0 ? void 0 : something.value) === true;
}
```

**Actual behavior:**
Actual Output:
```js
"use strict";
function isTrue(something) {
    var _a;
    return ((_a = something) === null || _a === void 0 ? void 0 : _a.value) === true;
}
```

**Comment by MartinJohns** [comment]
For simple cases this seems redundant, but for more complex cases this is necessary. In the case of properties there could be a getter with logic behind it, and in that case you wouldn't want it to be evaluated twice.

For example:

	class Fubar {
	    public get nested(): { value: boolean } {
	        console.log("nested accessed.");
	        return { value: true };
	    }
	}

	function isTrue(something: { nested: { value: boolean } | undefined }): boolean {
	    return something.nested?.value === true; 
	}

	isTrue(new Fubar());

Without the local variable you would get two console log statements, but with the local variable only one.

I would guess there is no special distinction whether this is necessary or not, and I would personally doubt the worth of such a check.

**Comment by bpasero** [comment]
Yeah I was not thinking it should be changed in all cases, only for the simple ones if possible.
